### PR TITLE
Add support for API Gateway as an EventBrigde target

### LINKED ISSUE #16624: Add API Gateway target support to aws_cloudwatch_event_target
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When creating rules in AWS Eventbridge, you can define targets. These targets can be other AWS services. One of the available target services is API Gateway. When selecting API Gateway in the AWS console, it is possible the set HTTP Headers and query string parameters, to define how the integration call will be made. One likely scenario is to set the `x-api-key` header, in case your API Gateway API requires an API key.

Currently, it looks like the `aws_cloudwatch_event_target` resource does not expose these properties.

### New or Affected Resource(s)

* aws_cloudwatch_event_target (Resource)

### References

* https://docs.aws.amazon.com/eventbridge/latest/userguide/targets-apigateway.html
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-target.html

### PR DESCRIPTION
**Pull request by rubenfonseca**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #16624

Output from acceptance testing:

```
$ make testacc TESTARGS="-run=TestAccAWSCloudWatchEventTarget_http"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventTarget_http -timeout 180m
=== RUN   TestAccAWSCloudWatchEventTarget_http
=== PAUSE TestAccAWSCloudWatchEventTarget_http
=== CONT  TestAccAWSCloudWatchEventTarget_http
--- PASS: TestAccAWSCloudWatchEventTarget_http (44.91s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	49.066s
```

### CODE REVIEW

File: website/docs/r/cloudwatch_event_target.html.markdown [hunk 1]
```diff
@@ -306,6 +339,7 @@ The following arguments are supported:
 * `batch_target` - (Optional) Parameters used when you are using the rule to invoke an Amazon Batch Job. Documented below. A maximum of 1 are allowed.
 * `kinesis_target` - (Optional) Parameters used when you are using the rule to invoke an Amazon Kinesis Stream. Documented below. A maximum of 1 are allowed.
 * `sqs_target` - (Optional) Parameters used when you are using the rule to invoke an Amazon SQS Queue. Documented below. A maximum of 1 are allowed.
+* `http_target` - (Optional) Parameters used when you are using the rule to invoke an API Gateway REST endpoint. Documented below. A maxium of 1 are allowed.
```
Comment (ewbankkit) [inline website/docs/r/cloudwatch_event_target.html.markdown:342]:
```suggestion
* `http_target` - (Optional) Parameters used when you are using the rule to invoke an API Gateway REST endpoint. Documented below. A maximum of 1 is allowed.
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchEventTarget_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventTarget_ -timeout 180m
=== RUN   TestAccAWSCloudWatchEventTarget_basic
=== PAUSE TestAccAWSCloudWatchEventTarget_basic
=== RUN   TestAccAWSCloudWatchEventTarget_EventBusName
=== PAUSE TestAccAWSCloudWatchEventTarget_EventBusName
=== RUN   TestAccAWSCloudWatchEventTarget_GeneratedTargetId
=== PAUSE TestAccAWSCloudWatchEventTarget_GeneratedTargetId
=== RUN   TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
=== PAUSE TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
=== RUN   TestAccAWSCloudWatchEventTarget_full
=== PAUSE TestAccAWSCloudWatchEventTarget_full
=== RUN   TestAccAWSCloudWatchEventTarget_disappears
=== PAUSE TestAccAWSCloudWatchEventTarget_disappears
=== RUN   TestAccAWSCloudWatchEventTarget_ssmDocument
=== PAUSE TestAccAWSCloudWatchEventTarget_ssmDocument
=== RUN   TestAccAWSCloudWatchEventTarget_http
=== PAUSE TestAccAWSCloudWatchEventTarget_http
=== RUN   TestAccAWSCloudWatchEventTarget_ecs
=== PAUSE TestAccAWSCloudWatchEventTarget_ecs
=== RUN   TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount
=== PAUSE TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount
=== RUN   TestAccAWSCloudWatchEventTarget_batch
=== PAUSE TestAccAWSCloudWatchEventTarget_batch
=== RUN   TestAccAWSCloudWatchEventTarget_kinesis
=== PAUSE TestAccAWSCloudWatchEventTarget_kinesis
=== RUN   TestAccAWSCloudWatchEventTarget_sqs
=== PAUSE TestAccAWSCloudWatchEventTarget_sqs
=== RUN   TestAccAWSCloudWatchEventTarget_input_transformer
=== PAUSE TestAccAWSCloudWatchEventTarget_input_transformer
=== RUN   TestAccAWSCloudWatchEventTarget_inputTransformerJsonString
=== PAUSE TestAccAWSCloudWatchEventTarget_inputTransformerJsonString
=== RUN   TestAccAWSCloudWatchEventTarget_PartnerEventBus
    resource_aws_cloudwatch_event_target_test.go:673: Environment variable EVENT_BRIDGE_PARTNER_EVENT_BUS_NAME is not set
--- SKIP: TestAccAWSCloudWatchEventTarget_PartnerEventBus (0.00s)
=== CONT  TestAccAWSCloudWatchEventTarget_basic
=== CONT  TestAccAWSCloudWatchEventTarget_ecs
=== CONT  TestAccAWSCloudWatchEventTarget_inputTransformerJsonString
=== CONT  TestAccAWSCloudWatchEventTarget_batch
=== CONT  TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount
=== CONT  TestAccAWSCloudWatchEventTarget_kinesis
=== CONT  TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
=== CONT  TestAccAWSCloudWatchEventTarget_full
=== CONT  TestAccAWSCloudWatchEventTarget_sqs
=== CONT  TestAccAWSCloudWatchEventTarget_ssmDocument
=== CONT  TestAccAWSCloudWatchEventTarget_GeneratedTargetId
=== CONT  TestAccAWSCloudWatchEventTarget_input_transformer
=== CONT  TestAccAWSCloudWatchEventTarget_disappears
=== CONT  TestAccAWSCloudWatchEventTarget_http
=== CONT  TestAccAWSCloudWatchEventTarget_EventBusName
--- PASS: TestAccAWSCloudWatchEventTarget_disappears (18.97s)
--- PASS: TestAccAWSCloudWatchEventTarget_sqs (19.97s)
--- PASS: TestAccAWSCloudWatchEventTarget_GeneratedTargetId (21.23s)
--- PASS: TestAccAWSCloudWatchEventTarget_ssmDocument (23.95s)
--- PASS: TestAccAWSCloudWatchEventTarget_http (27.48s)
--- PASS: TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount (37.63s)
--- PASS: TestAccAWSCloudWatchEventTarget_ecs (37.83s)
--- PASS: TestAccAWSCloudWatchEventTarget_EventBusName (38.91s)
--- PASS: TestAccAWSCloudWatchEventTarget_basic (44.32s)
--- PASS: TestAccAWSCloudWatchEventTarget_inputTransformerJsonString (46.24s)
--- PASS: TestAccAWSCloudWatchEventTarget_input_transformer (55.43s)
--- PASS: TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig (62.67s)
--- PASS: TestAccAWSCloudWatchEventTarget_kinesis (63.73s)
--- PASS: TestAccAWSCloudWatchEventTarget_full (64.03s)
--- PASS: TestAccAWSCloudWatchEventTarget_batch (79.20s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	82.461s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchEventTarget_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCloudWatchEventTarget_ -timeout 180m
=== RUN   TestAccAWSCloudWatchEventTarget_basic
=== PAUSE TestAccAWSCloudWatchEventTarget_basic
=== RUN   TestAccAWSCloudWatchEventTarget_EventBusName
=== PAUSE TestAccAWSCloudWatchEventTarget_EventBusName
=== RUN   TestAccAWSCloudWatchEventTarget_GeneratedTargetId
=== PAUSE TestAccAWSCloudWatchEventTarget_GeneratedTargetId
=== RUN   TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
=== PAUSE TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
=== RUN   TestAccAWSCloudWatchEventTarget_full
=== PAUSE TestAccAWSCloudWatchEventTarget_full
=== RUN   TestAccAWSCloudWatchEventTarget_disappears
=== PAUSE TestAccAWSCloudWatchEventTarget_disappears
=== RUN   TestAccAWSCloudWatchEventTarget_ssmDocument
=== PAUSE TestAccAWSCloudWatchEventTarget_ssmDocument
=== RUN   TestAccAWSCloudWatchEventTarget_http
=== PAUSE TestAccAWSCloudWatchEventTarget_http
=== RUN   TestAccAWSCloudWatchEventTarget_ecs
=== PAUSE TestAccAWSCloudWatchEventTarget_ecs
=== RUN   TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount
=== PAUSE TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount
=== RUN   TestAccAWSCloudWatchEventTarget_batch
=== PAUSE TestAccAWSCloudWatchEventTarget_batch
=== RUN   TestAccAWSCloudWatchEventTarget_kinesis
=== PAUSE TestAccAWSCloudWatchEventTarget_kinesis
=== RUN   TestAccAWSCloudWatchEventTarget_sqs
=== PAUSE TestAccAWSCloudWatchEventTarget_sqs
=== RUN   TestAccAWSCloudWatchEventTarget_input_transformer
=== PAUSE TestAccAWSCloudWatchEventTarget_input_transformer
=== RUN   TestAccAWSCloudWatchEventTarget_inputTransformerJsonString
=== PAUSE TestAccAWSCloudWatchEventTarget_inputTransformerJsonString
=== RUN   TestAccAWSCloudWatchEventTarget_PartnerEventBus
    resource_aws_cloudwatch_event_target_test.go:673: Environment variable EVENT_BRIDGE_PARTNER_EVENT_BUS_NAME is not set
--- SKIP: TestAccAWSCloudWatchEventTarget_PartnerEventBus (0.00s)
=== CONT  TestAccAWSCloudWatchEventTarget_basic
=== CONT  TestAccAWSCloudWatchEventTarget_ecs
=== CONT  TestAccAWSCloudWatchEventTarget_inputTransformerJsonString
=== CONT  TestAccAWSCloudWatchEventTarget_GeneratedTargetId
=== CONT  TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
=== CONT  TestAccAWSCloudWatchEventTarget_sqs
=== CONT  TestAccAWSCloudWatchEventTarget_kinesis
=== CONT  TestAccAWSCloudWatchEventTarget_batch
=== CONT  TestAccAWSCloudWatchEventTarget_input_transformer
=== CONT  TestAccAWSCloudWatchEventTarget_full
=== CONT  TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount
=== CONT  TestAccAWSCloudWatchEventTarget_disappears
=== CONT  TestAccAWSCloudWatchEventTarget_http
=== CONT  TestAccAWSCloudWatchEventTarget_ssmDocument
=== CONT  TestAccAWSCloudWatchEventTarget_EventBusName
=== CONT  TestAccAWSCloudWatchEventTarget_http
    provider_test.go:1115: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/12 10:21:28 [DEBUG] Using modified User-Agent: Terraform/0.12.30 HashiCorp-terraform-exec/0.13.3
        
        Error: Error creating API Gateway: BadRequestException: Endpoint Configuration type EDGE is not supported in this region: us-gov-west-1
        
          on terraform_plugin_test.tf line 25, in resource "aws_api_gateway_rest_api" "test":
          25: resource "aws_api_gateway_rest_api" "test" {
        
        
--- SKIP: TestAccAWSCloudWatchEventTarget_http (15.51s)
--- PASS: TestAccAWSCloudWatchEventTarget_disappears (22.77s)
--- PASS: TestAccAWSCloudWatchEventTarget_GeneratedTargetId (25.40s)
--- PASS: TestAccAWSCloudWatchEventTarget_sqs (25.98s)
--- PASS: TestAccAWSCloudWatchEventTarget_ssmDocument (29.15s)
--- PASS: TestAccAWSCloudWatchEventTarget_inputTransformerJsonString (39.36s)
--- PASS: TestAccAWSCloudWatchEventTarget_ecs (40.77s)
--- PASS: TestAccAWSCloudWatchEventTarget_ecsWithBlankTaskCount (41.01s)
=== CONT  TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig
    provider_test.go:1115: skipping test for aws-us-gov/us-gov-west-1: Error running apply: exit status 1
        2021/05/12 10:21:28 [DEBUG] Using modified User-Agent: Terraform/0.12.30 HashiCorp-terraform-exec/0.13.3
        
        Error: Creating CloudWatch Events Target failed: UnknownOperationException: Operation is disabled in this region.
        	status code: 400, request id: 78aaab31-c381-46d6-baec-85898092bfb1
        
          on terraform_plugin_test.tf line 54, in resource "aws_cloudwatch_event_target" "test":
          54: resource "aws_cloudwatch_event_target" "test" {
        
        
--- PASS: TestAccAWSCloudWatchEventTarget_EventBusName (45.35s)
--- PASS: TestAccAWSCloudWatchEventTarget_input_transformer (48.63s)
--- PASS: TestAccAWSCloudWatchEventTarget_basic (53.12s)
--- SKIP: TestAccAWSCloudWatchEventTarget_RetryPolicy_DeadLetterConfig (57.95s)
--- PASS: TestAccAWSCloudWatchEventTarget_kinesis (68.87s)
--- PASS: TestAccAWSCloudWatchEventTarget_full (69.49s)
--- PASS: TestAccAWSCloudWatchEventTarget_batch (141.53s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	144.461s
```

**Comment by ewbankkit** [comment]
@rubenfonseca Thanks for the contribution 👏.
In order to get this merged quickly I went ahead and addressed the linter errors and also modified the _flatten_ and _expand_ routines to match our [conventions](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md).
Other than that, everything was in great shape.

**Comment by rubenfonseca** [comment]
Thank you so much for your help! It was my first time with this code based, and I've already learned a lot about the conventions. Thank you!

**Pull request merged**